src/conversions/cart.cpp

```diff
@@ -121,8 +124,8 @@ static double geocentric_radius(double a, double b, double cosphi,
         This is from WP2, but uses hypot() for potentially better
         numerical robustness
     ***********************************************************************/
-    return hypot(a * a * cosphi, b * b * sinphi) /
-           hypot(a * cosphi, b * sinphi);
+    return hypot(a * a * cosphi, b *  b * sinphi) *
+       invhypot(a * cosphi, b * sinphi);
```

Comment:
I guess that will cause both a multiplication and a division to happen here...

Reply:
> I did try gathering a_cosphi and b_sinphi togther but (at least with -O3) that did not make a consistent improvement

No, because hopefully the optimizer will do that kind of folding even without being prompted to

Reply:
Just a quick comment: I think from looking at GCC -O3 -S output, as you suggest the compiler does indeed inline and yip you are right that in lines 127-128 multiply by invhypot() is indeed converted to division in the assembly code.  (GCC  version 10.2.1).